### DISCUSSION TITLE
Create an on chain deal status communication channel

### DISCUSSION THREAD

**Discussion by dd45e640b42e6da7da96faee3996ef7c** [filecoin-project/FIPs#246] [Enhancements - Technical]
The Storage Provider Working Groups NA call discussed a lot of legal issues regarding content and how to implement best practices regarding communication between storage clients and storage providers.

The basic idea is: 

Create a flag or an array of flags attached to the on chain status of the deal that the storage provider can update to communicate information on chain to a client he does not know, he cannot contact or when all other communication channels are blocked. 

The concrete use case that i brought up during the call (in the chat actually) is the canary in the coal mine for take down requests a storage provider receives and is legally not allowed to communicate to the client. But i can imagine other use cases as well where it is convenient to be able to communicate information via the only on chain connection client and storage provider have. 

I would suggest a bit array not bigger than 8 bits, that should allow for 256 states to be communicated.

**Comment by kaitlin-beegle** [comment]
Hi @f8-ptrk, just wanted to follow up on this proposal.  As a reminder, FIP drafts are usually authored by whomever sponsors a FIP discussion. 

Do you plan to write a FIP draft for this proposal? No problem if not, you're welcome to leave this as an open issue for the time being, but I wanted to check in. 

Thanks!

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
@kaitlin-beegle it's still under consideration to make this aFIP proposal. schedule is busy these days - i have it on the list - but it might take some weeks
